marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/reader/HubDataSourceReader.java

```diff
@@ -41,7 +50,12 @@ public HubDataSourceReader(DataSourceOptions dataSourceOptions) {
 
         this.options = dataSourceOptions.asMap();
         validateOptions(this.options);
+        this.hubClientConfig = Util.buildHubClientConfig(options);
+        this.hubClient = HubClient.withHubClientConfig(hubClientConfig);
+        this.objectMapper = new ObjectMapper();
+        this.customReadApiDefinitions = readCustomJobApiDefinitions(options);
 
+        this.endpointParams = determineReadRecordsEndpointParams(options);
```

Comment:
I think this is all work to be done in initializeRead, which is already instantiating a HubClient. That can become a local variable in initializeRead, which can then say - if you've specified an initializereadapipath option, then I'll read that using the HubClient and pass it in as the second arg to the SparkService.on call.